app/src/main/java/com/brainiacs/seandroidapp/TeacherDashboard/ButtonAdapter.java

```diff
@@ -113,10 +119,16 @@ public void onClick(DialogInterface dialog, int which) {
     //required amount of buttons
     //TODO
     public void setButtons(){
+        JSONTools jsonData = new JSONTools();
         GetClassesURLConnectionHandler classJSON = new GetClassesURLConnectionHandler("teacher/get-classes", "Data Retrieval Successful",
-                "Data Retrieval Failed", HttpURLConnectionHandler.Method.GET, null, mContext, null);
+                "Data Retrieval Failed", HttpURLConnectionHandler.Method.GET, null, mContext, null, jsonData);
         classJSON.execute((Void) null);
```

Comment:
This execute method doesn't call any handleResponse method in the HttpConnectionHandler or in the GetClassesURLConnectionHandler